### ISSUE TITLE
"List of indices into skeleton of end-effector constraints"

### ISSUE THREAD

**Issue by NPTP** [dilevin/computer-graphics-kinematics#35]
In `end_effectors_objective_and_gradient.cpp`, we have the the input `b` which represents "#b list of indices into skeleton of end-effector constraints".

I'm confused about this. `b` is a vector of ints, indexing into the skeleton. Indexing into the skeleton gives you a single bone. How does this also represent end-effector constraints? Are we talking about the constraints inside each Bone, and `b` presumably has less entries than `skeleton` since there are fewer end-effectors than there are bones?

Thanks!

**Comment by honglin-c** [comment]
Yes exactly. `b` is the list of indices into the skeleton of bones whose tips are constrained during IK, which is less than the number of bones in the skeleton in our cases.
